src/subdomains/generic/user/models/user/user.service.ts

```diff
@@ -564,4 +622,38 @@ export class UserService {
         }
       : { refCount: 0, refCountActive: 0 };
   }
+
+  // --- HELPER METHODS --- //
+  private async sendVerificationMail(userData: UserData, code: string, expirationMinutes: number): Promise<void> {
+    try {
+      if (userData.mail)
+        await this.notificationService.sendMail({
+          type: MailType.USER,
+          context: MailContext.EMAIL_VERIFICATION,
+          input: {
+            userData: userData,
+            title: `${MailTranslationKey.EMAIL_VERIFICATION}.title`,
+            salutation: {
+              key: `${MailTranslationKey.EMAIL_VERIFICATION}.salutation`,
+            },
+            suffix: [
+              {
+                key: `${MailTranslationKey.EMAIL_VERIFICATION}.message`,
+                params: { code },
+              },
+              { key: MailKey.SPACE, params: { value: '2' } },
+              {
+                key: `${MailTranslationKey.EMAIL_VERIFICATION}.closing`,
+                params: { expiration: `${expirationMinutes}` },
+              },
+              { key: MailKey.SPACE, params: { value: '4' } },
+              { key: MailKey.DFX_TEAM_CLOSING },
+            ],
+          },
+        });
+    } catch (e) {
+      this.logger.error(`Failed to send verification mail ${userData.id}:`, e);
+      throw new BadRequestException('Failed to send verification mail');
+    }
+  }
```

Comment:
I would remove the method here. It's nearly the same as `sendVerificationMail` in tfaService. I would use the method in tfaService, make it public and add a parameter to differentiate between EmailVerification and defaultVerification